### PR TITLE
removed unneccesary genericpcie path

### PR DESCRIPTION
**Pull request by sawanm-xilinx**
Fix: Removed unneccesary genericpciemodel path and added check for genericpciemodel binary executable permission.
Tests: Canary run. Results are clean.
Reviewer: sgundime, venkatp.

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -446,39 +446,26 @@ namespace xclswemuhal2
...
-#endif
+        std::string modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel";
 
         FILE *filep;
         if ((filep = fopen(modelDirectory.c_str(), "r")) != NULL)
         {
           // file exists
+          if (access(modelDirectory.c_str(), X_OK) != 0)
+          {
+            std::cout << "genericpciemodel binary does not have executable permission." << std::endl;
```
Comment (sgundime-xilinx) [inline src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx:461]: If executable permissions are not present then don't you think it needs to exit here?

File: src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx [hunk 2]
```diff
@@ -541,34 +541,22 @@ namespace xclswemuhal2 {
...
-#endif
+        std::string modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel";
 
         FILE *filep;
         if ((filep = fopen(modelDirectory.c_str(), "r")) != nullptr)
         {
           // file exists
+          if (access(modelDirectory.c_str(), X_OK) != 0)
+          {
+            std::cout << "genericpciemodel binary does not have executable permission." << std::endl;
```
Comment (sgundime-xilinx) [inline src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx:552]: If executable permissions are not present then don't you think it needs to exit here?

**Review by sgundime-xilinx — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx (hunk 1 above)
Comment (sawanm-xilinx) [reply src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx]: Yes, please check, updated as per suggestion.

File: src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx (hunk 2 above)
Comment (sawanm-xilinx) [reply src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx]: Yes, please check, updated as per suggestion.

File: src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx (hunk 1 above)
Comment (sgundime-xilinx) [reply src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx]: checked.

File: src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx (hunk 2 above)
Comment (sgundime-xilinx) [reply src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx]: checked.

**Review by sgundime-xilinx — APPROVED** [review]

**Review by venkatp-xilinx — APPROVED** [review]

**Comment by chvamshi-xilinx** [comment]
ok to test

**Comment by sgundime-xilinx** [comment]
> Build Failed! :(

retest please!

**Comment by chvamshi-xilinx** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by chvamshi-xilinx** [comment]
retest this please

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by xilinxgitops** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by sgundime-xilinx** [comment]
#7549 PR is adderssing the issue with a new branch. Please refer that PR. Thanks.

**Pull request closed without merging**